src/Symfony/Component/Translation/IdentityTranslator.php

```diff
@@ -20,19 +23,22 @@
  */
 class IdentityTranslator implements TranslatorInterface
 {
-    private $selector;
+    private $formatter;
     private $locale;
 
     /**
-     * Constructor.
-     *
-     * @param MessageSelector|null $selector The message selector for pluralization
+     * @param MessageFormatterInterface|MessageSelector $formatter The message formatter
      *
      * @api
      */
-    public function __construct(MessageSelector $selector = null)
+    public function __construct($formatter = null)
     {
-        $this->selector = $selector ?: new MessageSelector();
+        if ($formatter instanceof MessageSelector) {
+            @trigger_error('Passing a MessageSelector instance into the '.__METHOD__.' as a second argument is deprecated since version 2.8 and will be removed in 3.0. Inject a MessageFormatterInterface instance instead.', E_USER_DEPRECATED);
+            $formatter = new DefaultMessageFormatter($formatter);
+        }
```

Comment:
you should add an instanceof check to validate that the object is a ``MessageFormatterInterface``, as there is no typehint for BC reasons